controllers/movies.ts

```diff
@@ -0,0 +1,302 @@
+          if (movie.Type === "series") {
+            minutes = parseInt(movie.Runtime.split(" ")[0]);
+            let episodePerSeason = 10;
+            let seasons = 10;
+            total += episodePerSeason * seasons * minutes;
+          } else {
+            minutes = movie.Runtime.split(" ")[0];
+            total += +minutes;
+          }
+        }
+      }
+      console.log(total);
+      return total;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async getTotalVotes() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+      let total = 0;
+      for (const movie of movies) {
+        if (movie.imdbVotes !== "N/A") {
+          let votes = parseInt(movie.imdbVotes.replace(",", ""), 10);
+          total += +votes;
+        }
+      }
+      console.log(total);
+      return total;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async getImdbUrls() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+      const urlBase = "https://www.imdb.com/";
+      let urls = [];
+      for (const movie of movies) {
+        let movieUrl =
+          urlBase + movie.Title.replace(/\s/g, "") + "/" + movie.imdbID + "/";
+        urls.push(movieUrl);
+      }
+      return urls;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async getLanguages() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+
+      const languageMappings: { [key: string]: string } = {
+        English: "ENG",
+        Spanish: "ESP",
+      };
+
+      const shortenLanguages: string[] = [];
+
+      for (const movie of movies) {
+        const languages = movie.Language.split(",");
+
+        languages.forEach((language) => {
+          const trimmedLanguage = language.trim();
+          const shortenLanguage = languageMappings[trimmedLanguage] || "ENG";
+          console.log(shortenLanguage);
+          shortenLanguages.push(shortenLanguage);
+        });
+      }
+      return shortenLanguages;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async shortenUrl() {
+    try {
+      const movies = await Movie.find();
+      let shortenedUrls: String[] = [];
+
+      const shortenedURLs = await Promise.all(
+        movies.map(async (movie) => {
+          const randomImageIndex = Math.floor(
+            Math.random() * movie.Images.length
+          );
+          console.log(randomImageIndex);
+          const randomImageURL = await movie.Images[randomImageIndex];
+          console.log(randomImageURL);
+          const shortURL = await shortenURLs(randomImageURL);
+          console.log(shortURL);
+          shortenedUrls.push(shortURL);
+        })
+      );
+      return shortenedUrls;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+}
+export default Movies;
```

Comment:
This file has grown in rows and we can do something about it. Maybe get the helper functions and put them somewhere else. 🙂